pkg/controllers/vdb/upgrade.go

```diff
@@ -46,18 +46,20 @@ type UpgradeManager struct {
 	// Function that will check if the image policy allows for a type of upgrade (offline or online)
 	IsAllowedForUpgradePolicyFunc func(vdb *vapi.VerticaDB) bool
 	PrimaryImages                 []string // Known images in the primaries.  Should be of length 1 or 2.
+	SandboxName                   string   // sandbox we want to upgrade
 }
 
 // MakeUpgradeManager will construct a UpgradeManager object
 func MakeUpgradeManager(vdbrecon *VerticaDBReconciler, log logr.Logger, vdb *vapi.VerticaDB,
-	statusCondition string,
+	statusCondition, sandbox string,
 	isAllowedForUpgradePolicyFunc func(vdb *vapi.VerticaDB) bool) *UpgradeManager {
 	return &UpgradeManager{
 		VRec:                          vdbrecon,
 		Vdb:                           vdb,
 		Log:                           log,
 		Finder:                        iter.MakeSubclusterFinder(vdbrecon.Client, vdb),
 		StatusCondition:               statusCondition,
+		SandboxName:                   sandbox,
```

Comment:
I don't think setting sandbox for the entire upgrade manager makes sense. For replicated upgrade, it will span both the main cluster and a sandbox. So, it doesn't really make sense. I think each function in UpgradeManager that needs the sandbox should accept it as input.